sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -2899,6 +2899,21 @@ class AstBuilder(conf: SQLConf) extends SqlBaseBaseVisitor[AnyRef] with Logging
     }
   }
 
+  /**
+   * Create a [[DescribeNamespaceStatement]].
+   *
+   * For example:
+   * {{{
+   *   DESCRIBE DATABASE [EXTENDED] database;
```

Comment:
Since this PR changes the parser rule to accept `NAMESPACE`, we had better describe the change.
```
- DESCRIBE DATABASE [EXTENDED] database;
+ DESCRIBE (DATABASE|NAMESPACE) [EXTENDED] database;
```

Reply:
Maybe, the following is better?
```
DESCRIBE (DATABASE|SCHEMA|NAMESPACE) [EXTENDED] database;
```